app/controllers/svie_controller.rb

```diff
@@ -38,9 +38,7 @@ def hierarchy
   end
 
   def application_pdf
-    file_options = { filename: 'szerzodes.pdf', disposition: 'attachment', type: :pdf }
-    path = Rails.root.join('public', 'downloads', 'contract_empty.pdf')
-    send_file(path, file_options)
+    render html: (GenerateMembershipPdf.new(current_user).as_html).html_safe
```

Comment:
The html that we render here is not sanitized, vulnerable to XSS! I tried to sanitize the whole html, but it removes the link tag, therefore braking the page. I think its not a problem if we don't sanitize here, because everyone only sees their own page.

Reply:
You can sanitize in the user attributes in the `app/views/svie/pdf/_personal_data.html.erb` template and we should sanitize the group name too in the `app/views/svie/pdf/(outside_member|inside_member|inactive_member|not_member).html.erb.` template files.

Reply:
I looked it again. We can let the render method to sanitize all html. The problem with link tags has been solved, when I deleted the static links and added rails helpers.